app/scripts/controllers/metametrics.js

```diff
@@ -944,10 +955,36 @@ export default class MetaMetricsController {
         return resolve();
       };
 
-      this.segment.track(payload, callback);
+      this._submitSegmentEvent('track', payload, callback);
       if (flushImmediately) {
         this.segment.flush();
       }
     });
   }
+
+  // Method below submits the request to analytics SDK.
+  // For MV3 it will also add event to controller store
+  // and pass a callback to remove it from store once request is submitted to segment
+  // Saving events in controller store in MV3 ensures that events are tracked
+  // even if service worker terminates before events are submiteed to segment.
+  _submitSegmentEvent(eventType, payload, callback) {
```

Comment:
nit: I'd like to suggest renaming `_submitSegmentEvent` to `_submitSegmentAPICall`. In analytics land, events mean something different. For Segment events see: 
https://segment.com/docs/connections/spec/track/#event.

The suggestion to use "APICall" instead comes from https://segment.com/docs/connections/spec/

Reply:
👍 done